cpp/src/plasma/client.cc

```diff
@@ -503,7 +510,7 @@ Status PlasmaClient::Impl::GetBuffers(
             data + object->data_offset, object->data_size + object->metadata_size);
       } else {
 #ifdef PLASMA_CUDA
-        std::lock_guard<std::mutex> lock(gpu_mutex);
+        std::lock_guard<std::recursive_mutex> lock(gpu_mutex);
```

Comment:
It will cause compile error because of the definition "static std::mutex gpu_mutex;" at line 110.

Reply:
@shengjun1985 Thanks for catching this! It's fixed.